core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js

```diff
@@ -1,78 +1,90 @@
-function amCloser(e) {
-    var list = document.getElementById('visible-am-list');
-    var el = e.target;
-    while (el) {
-        if (el === list) {
-            return; // clicked in the list
+(function () {
+    function initializeAmMonitor(amMonitorRoot, options) {
+        var button = amMonitorRoot.querySelector('.am-button');
+        var url = button.getAttribute('data-href');
+        var amList = amMonitorRoot.querySelector('.am-list');
+
+        function onClose(e) {
+            var list = amList;
+            var el = e.target;
+            while (el) {
+                if (el === list) {
+                    return; // clicked in the list
+                }
+                el = el.parentElement;
+            }
+            close();
         }
-        el = el.parentElement;
-    }
-    hideVisibleAmList();
-}
-function secAmCloser(e) {
-    var list = document.getElementById('visible-sec-am-list');
-    var el = e.target;
-    while (el) {
-        if (el === list) {
-            return; // clicked in the list
+        function onEscClose(e) {
+            if (e.keyCode === 27) {
```

Comment:
would be good to extract this to a variable so that people who don't know keycodes know what this is

Reply:
I put a comment next to it to clarify, if that's OK.